lib/src/helpers/createClassNames.js

```diff
@@ -0,0 +1,12 @@
+import cx from 'classnames';
+
+function createClassNames(
```

Comment:
💚 I'm not 100% this new function is that helpful, isn't it essentially the same as using `cx`, but less flexible?

Reply:
My thinking was that pretty much every component should allow us to extend className.

Every time I add className as a prop I need to manually add propTypes, so what's the difference?

- using classnames: https://gist.github.com/kyleharper/c2d96476fdb4c94822e29682b31ef2e1

- using createClassNames: https://gist.github.com/kyleharper/1dacb3fecc1d931e6cb0f91655037f02

It's nearly a 50% reduction and if you consider that there are 100's of exported components in this library well let's do some math;

- 9 * 100 = 900 lines 
- 900 * 2 = 1800 lines
- 900 * 3 = 2700 lines

It actually saves quite a bit 👍 

-----------

The only question really is, how can we make the implementation better. I'm down for pairing on this if it makes it easier now that I have hopefully explained what I am trying to do here. 

Maybe the name could make it clear what we're saving on;

- `extendClassNameFromProps()`
- `extendCommonUseOfClassNames()`
- `combineClassNamesFromProps()`
- `createComponentClassNames()` (I quite like this one)

Reply:
Thanks for the in-depth explanation, I see the advantage!

The "using classnames: https://gist.github.com/kyleharper/c2d96476fdb4c94822e29682b31ef2e1" example is misleading in how much code is removed, as this example doesn't need `propTypes`.

It's also misleading to say that the two examples do different things with and without `defaultProps` - in both cases we pass `props.className` as undefined to `cx`. If we defaulted `props.className` to an empty string however, then I think that would be an advantage! But at that point, it is the only advantage of the function, and `cx` doesn't seem to mind being passed `cx` anyway.

My thinking is that both the examples below pass the same thing to `cx`, what's the advantage of the new function?

```javascript
import { createClassNames } from 'src/...';

function Component(props) {
  const classNames = createClassNames('component', props, {});
  
  return (
    <div clasName={classNames} />
  );
};

export { Component };
```

```javascript
import cx from 'classnames';

function Component(props) {
  const classNames = cx('component', props.className, {});
  
  return (
    <div clasName={classNames} />
  );
};

export { Component };
```

Reply:
Gone for `createComponentClassNames` and refined the names of the arguments too so it's easier to understand.